salt/auth/sqlite.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+'''
+Provide authentication using SQLite.
+
+When using SQLite as an authentication backend, you will need to create or
+use an existing table that has a username and a password column.
+
+To get started, create a simple table that holds just a username and
+a password. The password field will hold a SHA256 checksum (hex digest).
+
+.. code-block:: sql
+
+    CREATE TABLE `users` (
+      `id` INTEGER PRIMARY KEY ASC,
+      `username` VARCHAR(25),
+      `password` VARCHAR(64)
+    );
+
+To create a user within SQLite, execute the following statement.
+
+.. code-block:: sql
+
+    INSERT INTO users(username, password) VALUES ('diana',
+      '2bb80d537b1da3e38bd30361aa855686bde0eacd7162fef6a25fe97bf527a25b');
+
+Where username is `diana` and password is the SHA256 hex digest of `secret`.
+
+.. code-block:: yaml
+
+    sqlite_auth:
+      database: salt_users.db
+      username: root
+      password: letmein
+      auth_sql: 'SELECT username FROM users WHERE username = ? AND password = SHA256(?)'
+
+The `auth_sql` contains the SQL that will validate a user to ensure they are
+correctly authenticated. This is where you can specify other SQL queries to
+authenticate users.
+
+Enable SQLite authentication.
+
+.. code-block:: yaml
+
+    external_auth:
+      sqlite:
+        damian:
+          - test.*
+'''
+
+from __future__ import absolute_import
+import logging
+import hashlib
+import os.path
+
+log = logging.getLogger(__name__)
+
+
+def __get_connection_info():
+    '''
+    Grab SQLite Connection Details
+    '''
+    conn_info = {}
+    try:
+        conn_info['database'] = __opts__['sqlite_auth']['database']
+        conn_info['username'] = __opts__['sqlite_auth']['username']
+        conn_info['password'] = __opts__['sqlite_auth']['password']
+        conn_info['auth_sql'] = __opts__['sqlite_auth']['auth_sql']
+    except KeyError as e:
+        log.error('{0} does not exist'.format(e))
```

Comment:
Using ``str.format()`` is redundant in logging, as the logging module performs printf-style string replacement on the ``*args`` passed to it. Also, the log message here is ambiguous, it would basically say "database does not exist" or "username does not exist". This is not helpful at all in terms of troubleshooting. This message should be changed to:

```python
log.error('sqlite_auth is missing the \'%s\' configuration parameter', e)
```

We should probably also add a check before the try/except to make sure that the ``sqlite_auth`` is in the master opts at all. Something like this:

```python
if 'sqlite_auth' not in __opts__:
    log.error(
        'sqlite_auth configuration must be present in config file to use '
        'SQLite external authentication'
    )
    return None
```